src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,6 +1618,654 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
     ObjectGuid _playerGUID;
 };
 
+Position const LordaeronCancelScenePos = { 498.54f, 1560.2840f, 128.2032f, 1.100281f };
+
+enum ForsakenWarhorsePlayer
+{
+    NPC_FORSAKEN_WARHORSE_SYLVANAS              = 45057,
+
+    SPELL_RIDE_SYLVANAS_HORSE                   = 84166,
+    SPELL_LORDAERON_COMPLETE                    = 84185,
+    SPELL_DESPAWN_ALL_SUMMONS_LORDAERON         = 84173,
+    SPELL_FADE_TO_BLACK                         = 89092,
+
+    EVENT_ACTIVATE_SKIP                         = 1
+};
+
+// 45041 - Forsaken Warhorse (Player)
+struct npc_silverpine_warhorse_player_lordaeron : public ScriptedAI
+{
+    npc_silverpine_warhorse_player_lordaeron(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        me->HandleEmoteCommand(EMOTE_ONESHOT_MOUNT_SPECIAL);
+
+        me->CastSpell(nullptr, SPELL_RIDE_SYLVANAS_HORSE, true);
+
+        me->SetUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);
+        me->SetUnitFlag(UNIT_FLAG_IMMUNE_TO_NPC);
+
+        me->SetReactState(REACT_PASSIVE);
+    }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        _playerGUID = summoner->GetGUID();
+    }
+
+    void EnterEvadeMode(EvadeReason /*why*/) override { }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_ACTIVATE_SKIP:
+                {
+                    if (Player* player = ObjectAccessor::GetPlayer(*me, _playerGUID))
+                    {
+                        player->NearTeleportTo(LordaeronCancelScenePos, false);
+                        player->CastSpell(player, SPELL_DESPAWN_ALL_SUMMONS_LORDAERON, true);
+                    }
+                    break;
+                }
+                default:
+                    break;
+            }
+        }
+    }
+
+private:
+    EventMap _events;
+    ObjectGuid _playerGUID;
+};
+
+Position const LordaeronFinishPos = { 499.185f, 1549.9855f, 129.094f };
+
+enum ForsakenWarhorseSylvanas
+{
+    NPC_SYLVANAS_LORDAERON                      = 45051,
+    NPC_FORSAKEN_WARHORSE_PLAYER                = 45041,
+
+    PATH_WARHORSE_TO_SEPULCHER                  = 450570,
+
+    WAYPOINT_ARRIVED_TO_SEPULCHER               = 44,
+
+    POINT_CLOSE_TO_ZONE                         = 1,
+
+    SEAT_WARHORSE_SYLVANAS                      = 0
+};
+
+// 45057 - Forsaken Warhorse (Sylvanas)
+struct npc_silverpine_warhorse_sylvanas_lordaeron : public ScriptedAI
```

Comment:
looks like a script which could be moved to sai

Reply:
I really want to avoid moving anything related to sAI when it comes to RP events, especially because the mounting bug is still existing.